plugins/TKLiveSync/TKLiveSync/TKLiveSync.m

```diff
@@ -16,67 +19,172 @@ static void tryExtractLiveSyncArchive()
     NSString* libraryPath = [NSSearchPathForDirectoriesInDomains(NSLibraryDirectory, NSUserDomainMask, YES) firstObject];
     NSString* liveSyncPath = [NSString pathWithComponents:@[ libraryPath, @"Application Support", @"LiveSync" ]];
     NSString* syncZipPath = [NSString pathWithComponents:@[ liveSyncPath, @"sync.zip" ]];
-
-    if (![fileManager fileExistsAtPath:syncZipPath]) {
-        return;
-    }
+    NSString* appPath = [NSString pathWithComponents:@[ liveSyncPath, @"app" ]];
 
     NSError* err;
-
-    NSString* appPath = [NSString pathWithComponents:@[ liveSyncPath, @"app" ]];
-    if ([fileManager fileExistsAtPath:appPath]) {
-        [fileManager removeItemAtPath:appPath error:&err];
+    
+    if ([fileManager fileExistsAtPath:syncZipPath]) {
+        if ([fileManager fileExistsAtPath:appPath]) {
+            [fileManager removeItemAtPath:appPath error:&err];
+            if (err) {
+                NSLog(@"Can't remove %@: %@", appPath, err);
+            }
+        }
+        
+        NSLog(@"Unzipping LiveSync folder. This could take a while...");
+        NSDate* startDate = [NSDate date];
+        int64_t unzippedFilesCount = unzip(syncZipPath.UTF8String, liveSyncPath.UTF8String);
+        NSLog(@"Unzipped %lld entries in %fms.", unzippedFilesCount, -[startDate timeIntervalSinceNow] * 1000);
+        
+        [fileManager removeItemAtPath:syncZipPath error:&err];
         if (err) {
-            NSLog(@"Can't remove %@: %@", appPath, err);
+            NSLog(@"Can't remove %@: %@", syncZipPath, err);
         }
     }
 
-    NSLog(@"Unzipping LiveSync folder. This could take a while...");
-    NSDate* startDate = [NSDate date];
-    int64_t unzippedFilesCount = unzip(syncZipPath.UTF8String, liveSyncPath.UTF8String);
-    NSLog(@"Unzipped %lld entries in %fms.", unzippedFilesCount, -[startDate timeIntervalSinceNow] * 1000);
+    NSString* tnsModulesPath = [appPath stringByAppendingPathComponent:@"tns_modules"];
 
-    [fileManager removeItemAtPath:syncZipPath error:&err];
-    if (err) {
-        NSLog(@"Can't remove %@: %@", syncZipPath, err);
+    // TRICKY: Check if real dir tns_modules exists. If it does not, or it is a symlink, the symlink has to be recreated.
```

Comment:
It will work because we are [wiping out the whole `app` folder](https://github.com/NativeScript/ios-runtime/pull/747/files#diff-26b03154ad8867bf950c44b3bb6c9c86R28) in case a new zip archive is received.